client/src/components/text.zig

```diff
@@ -12,25 +12,80 @@ position: *usize,
 pub fn at(
     self: @This(),
     textBoxPosition: rl.Vector2,
+    textBoxSize: rl.Vector2,
 ) void {
-    return self.at_impl(0, textBoxPosition);
+    return self.at_impl(0, textBoxPosition, textBoxSize);
 }
 
 pub fn redacted_at(
     self: @This(),
     comptime buffer_size: usize,
     textBoxPosition: rl.Vector2,
+    textBoxSize: rl.Vector2,
 ) void {
     return if (buffer_size == 0)
         @compileError("Buffer size for redacted text must be non-zero")
     else
-        self.at_impl(buffer_size, textBoxPosition);
+        self.at_impl(buffer_size, textBoxPosition, textBoxSize);
+}
+
+fn pushCharacters(self: @This()) void {
+    var key = rl.getCharPressed();
+
+    while (key > 0 and key != 27 and key != 10) {
```

Comment:
Why is this a loop?

Reply:
Because in line 43 we are getting more characters.

Reply:
The problem is that while this loop is going on, the game loop is stopped.